MoreLinq.Test/TagFirstLastTest.cs

```diff
@@ -22,12 +22,29 @@ namespace MoreLinq.Test
     [TestFixture]
     public class TagFirstLastTest
     {
+        [Test]
+        public void TagFirstLastEvaluatesSourceLazily()
```

Comment:
Rename the test to be clear about why `source` has at least two elements.

```suggestion
        public void TagFirstLastDoesOneLookAhead()
```